### DISCUSSION TITLE
Getting same eTags for uploaded part using multipart upload to s3 bucket

### DISCUSSION THREAD

**Discussion by patimson** [aws/aws-sdk-java#2879] [Q&A]
Hi All,

I am trying to upload files in chunk using multiple part upload to s3 bucket. 

Results which I an see from logs:

implMultipartUploadService","message":"PartListing eTag...d41d8cd98f00b204e9800998ecf8427e PartNum: 1"
impl.MultipartUploadService","message":"PartListing eTag...d41d8cd98f00b204e9800998ecf8427e PartNum: 2"
impl.MultipartUploadService","message":"PartListing eTag...d41d8cd98f00b204e9800998ecf8427e PartNum: 3"

I am providing input for uploading parts as:
Request 1:
x-mc-file-id: 100
x-mc-file-name: abc1.pdf
x-mc-file-size: 19901641
x-mc-multipart-upload-id: AQAAAYSEKK-E1ccRLGZeRHKNtAfhhGFy2Wiqr6k
x-mc-part-num: 1

Request 2:
x-mc-file-id: 100
x-mc-file-name: abc1.pdf
x-mc-file-size: 19901641
x-mc-multipart-upload-id: AQAAAYSEKK-E1ccRLGZeRHKNtAfhhGFy2Wiqr6k
x-mc-part-num: 2

Request 3:
x-mc-file-id: 100
x-mc-file-name: abc1.pdf
x-mc-file-size: 19901641
x-mc-multipart-upload-id: AQAAAYSEKK-E1ccRLGZeRHKNtAfhhGFy2Wiqr6k
x-mc-part-num: 3

Request 4 as complete upload file part for last upload chunk:
x-mc-checksum: 3cd381dcdae793d89bd54a2aec7ac820
x-mc-file-id: 100
x-mc-file-size: 19901641
x-mc-multipart-upload-id: AQAAAYSEKK-E1ccRLGZeRHKNtAfhhGFy2Wiqr6k
x-mc-receiver: Test1
x-mc-sender: Test

Code:

UploadPartRequest partRequest =  partRequest.withBucketName(storeName).withInputStream(inputStream)
				.withKey("mmc-bucket").withPartNumber(request.getPartNum()).withUploadId(request.getUploadId())
				.withLastPart(false);
                            partRequest.getRequestClientOptions().setReadLimit(100000);
			
result = s3client.uploadPart(partRequest);

some of the values I have given as hard code values like file-id, sender , receiver. Note: I am not using sender, receiver anyway. so you can ignore that.

Please suggest your inputs what mistake am I doing here?
One more question: do we need file stream while uploading chunks to bucket? or we only need to give file-stream while last part of file-chunk?

**Comment by debora-ito** [comment]
@patimson apologies for the long delay in reply.

I don't quite understand the question. `x-mc-*` headers are not AWS headers.

Can you elaborate on the error you are seeing, or the problem you are trying to solve?
